glhe/profiles/external_base.py

```diff
@@ -1,24 +1,69 @@
-import pandas as pd
-from scipy.interpolate import interp1d
+from pathlib import Path
 
+import csv
+from datetime import datetime
+from glhe.utilities.functions import Interpolator1D
 
-class ExternalBase(object):
-
-    def __init__(self, path, col_num):
 
-        df = pd.read_csv(path, index_col=0, parse_dates=True)
-        df['delta t'] = df.index.to_series().diff().dt.total_seconds()
-        df['delta t'].iat[0] = 0
-        x_range = df['delta t'].cumsum().tolist()
-        y_range = df.iloc[:, col_num].tolist()
+class ExternalBase(object):
 
+    def __init__(self, path: Path, col_num: int):
+        x_range, y_range = self.read_csv_and_compute_delta_t(path, col_num)
         # added to allow multi-year simulations
         self.max_time = 0
-        x_range.append(x_range[-1] + (x_range[-1] - x_range[-2]))
-        y_range.append(y_range[0])
         self.max_time = x_range[-1]
+        self._interp_values = Interpolator1D(x_range, y_range)
+
+    @staticmethod
+    def read_csv_and_compute_delta_t(file_path: Path, col_num: int) -> [list, list]:
+        delta_t_values = []
+        y_values = []
+        last_timestamp = None
+
+        # Open CSV file and read data
+        with open(file_path, 'r') as csvfile:
+            csvreader = csv.reader(csvfile)
+            next(csvreader)  # Read header
+
+            for row in csvreader:
+                timestamp_str = row[0]
+
+                # Convert timestamp string to datetime object
+                current_timestamp = None
+                time_stamp_formats = ['%Y-%m-%d %H:%M:%S', '%m/%d/%Y %H:%M', '%m/%d/%Y %H:%M:%S']
```

Comment:
Pandas was happy to try some different time stamp formats it seems, because the actual run output was different from the unit test, and they both passed.  I just had to add a few options here.  I think it would be sufficient to modify the unit test to match the run output, and then we just have our specific format.  But this is also fine.